guides/theming.md

```diff
@@ -1,311 +1,359 @@
+Material. Unless your application uses every single component, this will produce unecessary CSS.
+
+```scss
+@use '~@angular/material' as mat;
 
+$my-primary: mat.define-palette(mat.$indigo-palette, 500);
+$my-accent: mat.define-palette(mat.$pink-palette, A200, A100, A400);
 
-// Define an alternate dark theme.
-$dark-primary: mat-palette($mat-blue-grey);
-$dark-accent:  mat-palette($mat-amber, A200, A100, A400);
-$dark-warn:    mat-palette($mat-deep-orange);
-$dark-theme:   mat-dark-theme((
-  color: (
-    primary: $dark-primary,
-    accent: $dark-accent,
-    warn: $dark-warn,
-  )
+$my-theme: mat.define-light-theme((
+ color: (
+   primary: $my-primary,
+   accent: $my-accent,
+ )
 ));
 
-// Include the dark color styles inside of a block with a CSS class. You can make this
-// CSS class whatever you want. In this example, any component inside of an element with
-// `.unicorn-dark-theme` will be affected by this alternate dark theme instead of the default theme.
-.unicorn-dark-theme {
-  @include angular-material-color($dark-theme);
-}
+@include mat.all-component-themes($my-theme);
 ```
 
-In the above example, any component inside a parent with the `unicorn-dark-theme` class will use
-the dark theme, while other components will fall back to the default `$candy-app-theme`.
+To include the emitted styles in your application, [add your theme file to the `styles` array of
+your project's `angular.json` file][adding-styles].
 
-You can include as many color schemes as you like in this manner. You can also `@include` the
-`angular-material-color` in separate files and then lazily load them based on an end-user
-interaction (how to lazily load the CSS assets will vary based on your application).
+[adding-styles]: https://angular.io/guide/workspace-config#styles-and-scripts-configuration
 
-It's important to remember, however, that the `mat-core` mixin should only ever be included _once_.
-Similarly, the `angular-material-theme` mixin should not be used multiple times as it generates
-styles for all configured theming system parts. For example, typography styles would be generated
-multiple times, even though the configuration did not change. Instead, use fine-grained mixins such
-as `angular-material-color` that only result in styles being generated for the [color system][2].
+### Using a pre-built theme
 
-Read more about duplicated theme styles in the [dedicated guide](./duplicate-theming-styles.md).
+Angular Material includes four pre-built theme CSS files, each with different palettes selected.
+You can use one of these pre-built themes if you don't want to define a custom theme with Sass.
 
-#### Multiple themes and overlay-based components
+| Theme                  | Light or dark? | Palettes (primary, accent, warn) |
+|------------------------|----------------|----------------------------------|
+| `deeppurple-amber.css` | Dark           | deep-purple, amber, red          |
```

Comment:
can we map this to look like the example of how we explain defining themes?
I think that might remove some of the mystery of these pre-defined themes

```scss
$deeppurple-amber: mat.define-dark-theme((
 color: (
   primary: mat.define-palette(mat.$deep-purple-palette, ???),
   accent: mat.define-palette(mat.$amber-palette, ???)
 )
));
```

Reply:
I added a sentence that links to the source so people can reference these implementations as examples.